includes/class-newspack-image-credits.php

```diff
@@ -0,0 +1,430 @@
+<?php
+/**
+ * Image credits for media library items.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+/**
+ * Newspack_Image_Credits class.
+ */
+class Newspack_Image_Credits {
+
+	/**
+	 * Meta key for the credit.
+	 */
+	const MEDIA_CREDIT_META = '_media_credit';
+
+	/**
+	 * Meta key for the credit URL.
+	 */
+	const MEDIA_CREDIT_URL_META = '_media_credit_url';
+
+	/**
+	 * Meta key for the credit organization.
+	 */
+	const MEDIA_CREDIT_ORG_META = '_navis_media_credit_org';
+
+	/**
+	 * Meta key for whether the image is distributable.
+	 */
+	const MEDIA_CREDIT_CAN_DISTRIBUTE_META = '_navis_media_can_distribute';
+
+	/**
+	 * Hook actions and filters.
+	 */
+	public static function init() {
+		add_filter( 'attachment_fields_to_save', [ __CLASS__, 'save_media_credit' ], 10, 2 );
+		add_filter( 'attachment_fields_to_edit', [ __CLASS__, 'add_media_credit' ], 10, 2 );
+		add_filter( 'get_the_excerpt', [ __CLASS__, 'add_credit_to_attachment_excerpts' ], 10, 2 );
+		add_filter( 'render_block', [ __CLASS__, 'add_credit_to_image_block' ], 10, 2 );
+		add_filter( 'wp_get_attachment_image_src', [ __CLASS__, 'maybe_show_placeholder_image' ], 11, 4 );
```

Comment:
The placeholder image applies to the site icon, too. This image "placement" should not be affected by the replacement.

Reply:
I wasn't sure about that, but I think you're probably right since no site should be using a non-licensed third-party image as their site logo. 😆 Fixed in d2376bf but fixed the wonky logic in 07604e3.